frontend/notarial-blockchain-ui/src/App.tsx

```diff
@@ -1,28 +1,152 @@
-import './App.css';
-import Button from '@mui/material/Button';
+import "./App.css";
 
-const  App = () => {
+import { CircularProgress } from "@material-ui/core";
+import Button from "@mui/material/Button";
+import axios from "axios";
+import React, { useCallback, useState } from "react";
+
+type ConnectionType = "NOT_CONNECTED" | "CONNECTING_ERROR" | "CONNECTED";
+type User = {
+  port?: string;
+  ip?: string;
+};
+
+const App = () => {
+  const [connected, setConnected] = useState<ConnectionType>("NOT_CONNECTED");
+  const [loading, setLoading] = useState(false);
+  const [user, setUserData] = useState<User>({});
+  const [message, setMessage] = useState<string[]>([]);
+
+  const connect = () => {
+    setLoading(true);
+    setMessage((prevState) => ["Connecting", ...prevState.slice(0, 6)]);
+    axios
+      .post("http://localhost:8080/node/connect")
```

Comment:
Maybe we should create a url builder, because if we want to change the domain (e.g. external server, we have to change each line in code). Some other class/handler should be responsible for that, right?

Reply:
good point, will add `TODO`